src/org/labkey/test/tests/component/QueryGridTest.java

```diff
@@ -0,0 +1,443 @@
+package org.labkey.test.tests.component;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.TestTimeoutException;
+import org.labkey.test.categories.InDevelopment;
+import org.labkey.test.components.glassLibrary.grids.QueryGrid;
+import org.labkey.test.pages.test.CoreComponentsTestPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.SampleTypeDefinition;
+import org.labkey.test.util.TestDataGenerator;
+import org.labkey.test.util.exp.SampleTypeAPIHelper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+@Category({InDevelopment.class})
+public class QueryGridTest extends BaseWebDriverTest
+{
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        QueryGridTest init = (QueryGridTest) getCurrentTest();
+
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+    }
+
+    @Before
+    public void preTest() throws Exception
+    {
+        goToProjectHome();
+    }
+
```

Comment:
This seems unnecessary (I should probably remove it from the template).
All of the tests start with a `CoreComponentsTestPage.beginAt`